CMakeLists.txt

```diff
@@ -0,0 +1,107 @@
+# Copyright (c) 2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+# IMPORTANT: Changes which affect binary results may not be quietly gated
+#            by CMake version.
+#
+# Debian 10 Buster, https://wiki.debian.org/LTS, EOL 2024:
+#  - CMake 3.13.4, https://packages.debian.org/buster/cmake
+#
+# Ubuntu 22.04 Jammy, https://wiki.ubuntu.com/Releases, EOL 2032:
+#  - CMake 3.22.1, https://packages.ubuntu.com/jammy/cmake
+#
+# Visual Studio 17 2022, https://visualstudio.microsoft.com:
+#  - CMake 3.24
+#
+# All policies known to the running version of CMake and introduced
+# in the 3.24 version or earlier will be set to use NEW behavior.
+# All policies introduced in later versions will be unset.
+# See: https://cmake.org/cmake/help/latest/manual/cmake-policies.7.html
+cmake_minimum_required(VERSION 3.13...3.24)
+
+project("Bitcoin Core"
+  VERSION 24.99.0
+  DESCRIPTION "Bitcoin client software"
+  HOMEPAGE_URL "https://bitcoincore.org/"
+  LANGUAGES CXX C ASM
+)
+
+set(CLIENT_VERSION_IS_RELEASE "false")
+set(COPYRIGHT_YEAR "2023")
+set(COPYRIGHT_HOLDERS "The %s developers")
+set(COPYRIGHT_HOLDERS_FINAL "The ${PROJECT_NAME} developers")
+set(PACKAGE_BUGREPORT "https://github.com/bitcoin/bitcoin/issues")
+
+# Configurable options.
+# When adding a new option, end the <help_text> with a full stop for consistency.
+include(CMakeDependentOption)
+cmake_dependent_option(CXX20 "Enable compilation in C++20 mode." OFF "NOT MSVC" ON)
+
+if(CXX20)
+  set(CMAKE_CXX_STANDARD 20)
+else()
+  set(CMAKE_CXX_STANDARD 17)
+endif()
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+set(configure_warnings)
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.14)
+  include(CheckPIESupported)
+  check_pie_supported(OUTPUT_VARIABLE check_pie_output LANGUAGES CXX)
+  if(NOT CMAKE_CXX_LINK_PIE_SUPPORTED)
+    list(APPEND configure_warnings "PIE link options are not supported for executable targets: ${check_pie_output}.")
+  endif()
+else()
```

Comment:
Sorry I missed this before ACKing, but I'm afraid this is exactly what I was talking about here: https://github.com/bitcoin/bitcoin/pull/27060#discussion_r1104910467

AFAICS, building older CMake will result in an inferior link line. I think this one's ok because build will probably fail without it, but this is a slippery slope to "newer CMake == better binaries".

Reply:
Do you have any particular approach in mind?

I'm sure, in this case, it is possible to emulate CMake 3.14+ behavior when running CMake 3.13.